src/oss-tests/ProjectManagerTests/ProjectManagerFactoryTests.cs

```diff
@@ -127,16 +130,70 @@ public void RemoveAllProjectManagersSucceeds()
         _managerOverrides.Clear();
 
         ProjectManagerFactory projectManagerFactory = new(_managerOverrides);
-        
+
         AssertFactoryCreatesCorrect(projectManagerFactory);
-        
+
         foreach (PackageURL packageUrl in ProjectManagerFactory.GetDefaultManagers(_httpClientFactory).Keys
                      .Select(purlType => new PackageURL($"pkg:{purlType}/foo")))
         {
             Assert.IsNull(projectManagerFactory.CreateProjectManager(packageUrl));
         }
     }
 
+    /// <summary>
+    /// Test that timeout is set if a value is passed.
+    /// </summary>
+    [TestMethod]
+    public void CreateProjectManagerSetsTimeOutCorrectly()
+    {
+        // Arrange
+        ProjectManagerFactory projectManagerFactory = new();
+        TimeSpan testTimeout = TimeSpan.FromMilliseconds(100);
+        PackageURL testPackageUrl = new("pkg:npm/foo");
+
+        // Act
+        IBaseProjectManager? testProjectManager = projectManagerFactory.CreateProjectManager(testPackageUrl, ".", testTimeout);
+        IBaseProjectManager? testProjectManagerWithoutTimeout = projectManagerFactory.CreateProjectManager(testPackageUrl, ".");
+
+        // Assert
+        Assert.AreEqual(testTimeout, testProjectManager?.Timeout);
+        Assert.IsNull(testProjectManagerWithoutTimeout?.Timeout);
+    }
+
+    /// <summary>
+    /// Test that requests time out after the timespan if specified.
+    /// </summary>
+    [TestMethod]
+    public async Task PackageManagerRequestsTimeOutCorrectly()
+    {
+        // Arrange
+        Mock<IHttpClientFactory> mockFactory = new();
+        MockHttpMessageHandler mockHttp = new();
+        mockHttp
+            .When(HttpMethod.Get, "*")
+            .Respond(async () =>
+            {
+                await Task.Delay(5000); // simulate a 5 seconds delay
```

Comment:
Whenever we delay using wall-clock time in unit tests, that can be pretty annoying (since developers sit through the delay whenever the tests run). (The best way to check timeouts in unit tests would be to simulate the passage of time by mocking a time provider interface, so that unit tests don't actually wait at all - but since mockHTTP is integrating with built-in .NET HTTP classes, that doesn't seem to be an option unfortunately).

I see that the timeout you specified is much smaller (100 ms); could this delay be 120ms?

Reply:
Yes, good catch I meant to have that as a smaller delay here. Updated it to be 120ms.

Reply:
Looks great, thanks!